public/components/utils/ConfirmButton.react.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+
+export default class ConfirmButton extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      clicked: false
+    };
+
+    this.resetClick = this.resetClick.bind(this);
+  }
+
+  onFirstClick() {
+    this.setState({
+      clicked: true
+    });
+
+    setTimeout(this.resetClick, 2000);
+  }
+
+  onSecondClick(e) {
+    this.props.onClick(e);
+
+    clearTimeout(this.resetClick);
+  }
+
+  resetClick() {
+    this.setState({
+      clicked: false
+    });
+  }
+
+  render () {
+
+    if (this.state.clicked) {
+      return (
+        <button className={this.props.className} onClick={this.onSecondClick.bind(this)}>Are You Sure? Click Again To Confirm</button>
```

Comment:
A bit pedantic but can't the text just Confirm? or something. Seems a bit wordy that's all.